urwid/main_loop.py

```diff
@@ -1324,7 +1324,10 @@ def run(self):
         self._loop.set_exception_handler(self._exception_handler)
         self._loop.run_forever()
         if self._exc_info:
-            raise self._exc_info[0](self._exc_info[1]).with_traceback(self._exc_info[2])
+            if PYTHON3:
+                raise self._exc_info
+            else:
+                raise self._exc_info[0](self._exc_info[1]).with_traceback(self._exc_info[2])
```

Comment:
`__super` works fine, but Python 3 in general won't (necessarily) work on a fresh checkout, because urwid still relies on 2to3.  `__super` is set by a metaclass, which is only set by `__metaclass__`, which is just a regular class attribute in Python 3.

Run the tests with `python3 setup.py test`.  That'll do an sdist build (which will run 2to3) and run the tests from within the build.